sites/next.skeleton.dev/src/content/docs/resources/contribute/components.mdx

```diff
@@ -221,6 +221,43 @@ setSession({ user: 'bar' });
 const session = getSession();
 ```
 
+## Form Components
+
+When implementing form components (ex: Segment Control), please adhere to the following conventions:
+
+1. For components with a parent/child structure, pass all shared values through the parent.
+2. Data in should use a `value` prop, while data out should use an `onChange` event handler.
+3. When supported, use two-way binding on the `value`; `onChange` is then optional.
+4. Pass state that's unique to the child components directly to each child item.
+5. When embedding form inputs within components for state management, always use `type="hidden"`.
+
+See the examples per each framework below.
+
+### Svelte
+
+```ts
+const value = $state('foo');
+```
+```svelte
+<!-- onChange is option and not required -->
+<Control bind:value={value} name="flavors">
```

Comment:
In Svelte you can just do `bind:value` because the prop and variable name are identical. Not sure if we want to promote these shorthands, but less code is probably better.

Reply:
Svelte and/or Prettier is auto-formatting to shorthand for us, always happens on save for me in most cases. So if we want to change that we'll need to modify this configuration.

I opted for the more verbose option to explain the concept.